### PR TITLE
[incubator/orientdb] Support for existing volumeClaim or volumeClaimTemplates & storage classes with volumeClaimTemplates

### PR DESCRIPTION
**Pull request by muthufmass**
#### What this PR does / why we need it:
To support existing volume claim with the chart for storage and backup folders to stay with persistence storage. second to support storage class name in autocreating volume claims through the chart while setting up orient. This will help in mapping NFS storage or other specific storage classes.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

**Comment by muthufmass** [comment]
I didnt do DCO on each commit. As this is my first time giving pull request in this repo. I can go and change now. So what should I do, keep me posted...

File: incubator/orientdb/templates/statefulset.yaml [hunk 1]
```diff
@@ -122,23 +124,41 @@ spec:
             name: {{ template "orientdb.fullname" . }}-configmap
         - name: orientdb-config-vol
           emptyDir: {}
+  {{- if (.Values.persistence.enabled) and (.Values.persistence.volumeClaim) }}
+      {{- if .Values.persistence.volumeClaim.storageName }}
```
Comment (zanhsieh) [inline incubator/orientdb/templates/statefulset.yaml:128]: template: orientdb/templates/statefulset.yaml:128:20: executing "orientdb/templates/statefulset.yaml" at <.Values.persistence....>: can't evaluate field storageName in type interface {}

complained by circleci-chart-lint

Comment (muthufmass) [reply incubator/orientdb/templates/statefulset.yaml:128]: I have extended the same and may not be needed for everyone with existing volumeclaim, so have commented .Values.persistence.volumeClaim.storageName in values.yaml file. However if line 127 is not satisfied, it shouldn't get into the 128th line. Correct me if I am wrong here.

Comment (zanhsieh) [reply incubator/orientdb/templates/statefulset.yaml:128]: Line 127 itself basically is wrong. It should be 

'''
{{- if and (.Values.persistence.enabled) (.Values.persistence.volumeClaim) }}
'''

Comment (muthufmass) [reply incubator/orientdb/templates/statefulset.yaml:128]: Got It! I will fix it properly and resubmit

**Pull request closed without merging**